### PR TITLE
Fix measurements for case-insensitive Azure 

### PR DESCRIPTION
**Pull request by Nirusu**
### Proposed change(s)

- Fix measurements for case-insensitive Azure
- Fix CI pipeline in general (I believe, main is broken so cannot test...)

Since the AzureRM terraform provider enforces certain spelling according to their arbitrary standards, we added image name normalisation to make this process a bit user-friendlier. However, since S3 is case-sensitive, we also need to normalise file names there, too. 

As a stop-gap measure before we completely rework the measurement fetching anyway, this PR does that plus fixes the pipeline which was broken before. 

It might still be broken as my test currently dies at the verification-service. Hope main is fixed soon so I can re-base and build a new image any try again since I don't like merging untested code.

### Checklist

- [ ] Update [CHANGELOG.md](https://github.com/edgelesssys/constellation/blob/main/CHANGELOG.md)
- [ ] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [ ] Link to Milestone

### CODE REVIEW

**Comment by Nirusu** [comment]
✅ Action runs: https://github.com/edgelesssys/constellation/actions/runs/3364635553
✅ fetch-measurements works on testing (when replacing the prod public key with the new dev public key)

Should be ready for review.

**Review by daniel-weisse — CHANGES_REQUESTED** [review]

File: cli/internal/cmd/upgradeplan.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ import (
 const imageReleaseURL = "https://github.com/edgelesssys/constellation/releases/latest/download/versions-manifest.json"
 
 var (
-	azureCVMRxp = regexp.MustCompile(`^\/CommunityGalleries\/ConstellationCVM-b3782fa0-0df7-4f2f-963e-fc7fc42663df\/Images\/constellation\/Versions\/[\d]+.[\d]+.[\d]+$`)
+	azureCVMRxp = regexp.MustCompile(`^(?i)\/CommunityGalleries\/ConstellationCVM-b3782fa0-0df7-4f2f-963e-fc7fc42663df\/Images\/constellation\/Versions\/[\d]+.[\d]+.[\d]+$`)
```
[the remark this answers is not in the document]
Comment (malt3) [reply cli/internal/cmd/upgradeplan.go:36]: This is only for the warning `image %q does not look like a released production image for Azure` so it should be fine.

File: .github/workflows/generate-measurements.yml [hunk 2]
```diff
@@ -20,6 +20,12 @@ on:
         type: boolean
         required: true
 
+env:
+  ARM_CLIENT_ID: "b657a00e-813a-4dc7-9b09-fa498a254d71"
+  ARM_CLIENT_SECRET: ${{ secrets.AZURE_E2E_CLIENT_SECRET }}
+  ARM_SUBSCRIPTION_ID: "0d202bbb-4fa7-4af8-8125-58c269a05435"
+  ARM_TENANT_ID: "adb650a8-5da3-4b15-b4b0-3daf65ff7626"
```
[the remark this answers is not in the document]
Comment (Nirusu) [reply .github/workflows/generate-measurements.yml:27]: True... I mean they are leaked anyways by now from the other actions (shouldn't be sensitive though) but I'll switch them to secrets so they are at least replaceable easily.

[the remark this answers is not in the document]
Comment (Nirusu) [reply .github/workflows/generate-measurements.yml:27]: Changed it, also replaced them in the E2E test so they should hopefully work in general.

**Comment by Nirusu** [comment]
Action works now (for Azure). Let me know what you think about it @daniel-weisse

**Review by malt3 — APPROVED** [review]
LGTM

File: .github/workflows/generate-measurements.yml (hunk 2 above)
Comment (daniel-weisse) [inline .github/workflows/generate-measurements.yml:27]: Can be removed now, since the values are injected via secrets

**Review by daniel-weisse — APPROVED** [review]

File: .github/workflows/generate-measurements.yml (hunk 2 above)
Comment (Nirusu) [reply .github/workflows/generate-measurements.yml]: Oops, how did I forget this one... Fixed, thanks!

**Pull request merged**